### PR TITLE
Use FAST_RUN for reference version.

### PR DESCRIPTION
**Pull request by lamblin**
This should speed up the FAST_COMPILE buildbot

### CODE REVIEW

**Comment by abergeron** [comment]
I don't know.  It doesn't seem like either is the reference version, but more like they are testing each other.

**Comment by lamblin** [comment]
"full" and "half" have been added recently, and it looks like the tests are building a more complex graph (explicit padding) to be able to use "valid".
@botev can you confirm?

**Comment by botev** [comment]
Yes, the tests for "full" and "half" compare against the explicitly padded input and then running "valid". This was so that the testing is less prone to my own design/choice of tests and relies on already well-tested implementation.

**Comment by lamblin** [comment]
OK, thanks. So using the C version of "valid" when testing the Python version of "full" and "half" would be reasonable in that case.

File: theano/tensor/nnet/tests/test_neighbours.py [hunk 1]
```diff
@@ -283,9 +283,9 @@ def test_neibs_full_step_by_valid(self):
                     padded_x = T.set_subtensor(padded_x[:, :, extra[0]:-extra[0], extra[1]:-extra[1]], x)
                     x_using_valid = images2neibs(padded_x, neib_shape, neib_step, mode="valid")
                     x_using_full = images2neibs(x, neib_shape, neib_step, mode="full")
-                    close = T.allclose(x_using_valid, x_using_full)
-                    f = theano.function([], close, mode=self.mode)
-                    assert f()
+                    f_valid = theano.function([], x_using_valid, mode='FAST_RUN')
+                    f_full = theano.function([], x_using_full, mode=self.mode)
+                    unittest_tools.assert_allclose(f_valid(), f_full())
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_neighbours.py]: Did you time that this make a difference in the test time? This lower a little bit the test coverage as DebugMode isn't used. But I think think this is a problem. If you see test time difference on your computer, I'm ok to merge this PR.

Comment (lamblin) [inline theano/tensor/nnet/tests/test_neighbours.py]: The tests in question goes from around 8 minutes to around 4, I think.

Comment (lamblin) [inline theano/tensor/nnet/tests/test_neighbours.py]: There are already tests for "valid" that are executed in DebugMode.

**Pull request merged**